### PR TITLE
Shooter game

### PR DESCRIPTION
**Pull request by uryelah**
**This is the PR for the '**From the depths...**' HTML5 game as part of the Javascript capstone project submission.**

It is playable online [here](https://uryelah.github.io/from-the-depths/) and consists of controlling a squid-like tentacle undersea. 

you

The player must shot squeeze humans dead as they invade its ocean and cause hp damage.

The game is completely controlled with the mouse.

**Initial objectives:**

1. At the beginning of the project, I wanted to use the Geometry tools of Phaser to create dynamic joints that would wrap and rotate nicely as one partitioned line representing the player tentacle. 
- [ACHIEVED} I managed to make the player controllable character a tentacle and have it moving in a wiggly way, following the mouse and behaving like it has an inner joints structure.
- [NOT ACHIEVED]This proved to be too complex, math-wise, and I wasted one day trying to make it work with very doubtful results. In the end, I decided to approach it in a different way using child containers.

2. I wanted to learn and use the framework to its fullest and work with well-contained classes and functions.
- [ACHIEVED] I did learn a lot about the actual framework while working on it, especially after day 3.
- [NOT ACHIEVED] In the end, it would take too much time to get a deeper understanding of the Phaser API so I went in with a lot of regular Vanilla Javascript.
I also got carried away with the issues referred to at item 1 and while trying to make the features work got sloppy with the code quality and encapsulation.

3. I planned on delivering the project on Thursday
- [NOT ACHIEVED] This was not achieved due to the need of fixing accumulated tech debts that would make the code un-parsable.

4. I wanted to make a game with a subject that was not very generic or too similar to the tutorial.
- [ACHIEVED] I'm happy about the 2 days spent ideating alternatives to the good old shooter. I also like the undersea idea contributed to a lot of the actual player experience and challenged me technically on its implementation

5. I wanted to make this project a well-finished game so that I can show it in my portfolio.
- [PARTIALLY ACHIEVED] While I did put some effort on the ambiance, sound and graphics of the game I still think it has room for improvement. As of right now, the game is something that I can be moderately proud to show.

6. I wanted to have a well-delimited business and interface logic or to use the entity-component-system paradigm.
- [NOT ACHIEVED] Again, I got carried away testing the moving tentacle physics and the framework and did not plan the structure as well as I would have wanted to.

7. There were a lot of features that I planned on implementing but was overly optimistic with the required time, such as levels, helper characters, a boss fight, and power-ups.
- [PARTIALLY ACHIEVED] I managed to implement a rudimentary level, or difficulty rise by time, system and different enemy characters.

Basically, I had a lot of problems regarding time management and getting too obsessed with smaller details instead of taking more time to make the code more maintainable.

submarine

I learned a lot and plan to keep working on this project on the weekends.

### CODE REVIEW

**Comment by uryelah** [comment]
- Added some test coverage to the helper functions
- Improved the read me game documentation
 Please check the shooter-game branch README file for the complete readme file, not the master branch.
